### PR TITLE
Refactored web/mod.rs

### PR DESCRIPTION
**Pull request by Kixiron**
* Removed redundant `'static` lifetimes from constants
* Method chain formatting
* Transferred from the painful if statement to a wonderful match
* Overall style and formatting changes

### CODE REVIEW

**Comment by jyn514** [comment]
Can you modify this PR so it doesn't change the formatting? If so I think it would be useful, especially fixing the clippy lints and the `match` instead of the `if/else` chains.

File: src/web/mod.rs [hunk 1]
```diff
@@ -402,25 +398,26 @@ impl Server {
...
+    match delta {
+        (days, _, _, _) if days > 5 => format!("{}", tm.strftime("%b %d, %Y").unwrap()),
+        (days, _, _, _) if days > 1 => format!("{} days ago", days),
+        (days, _, _, _) if days == 1 => "one day ago".to_string(),
+
+        (_, hours, _, _) if hours > 1 => format!("{} hours ago", hours),
+        (_, hours, _, _) if hours == 1 => "an hour ago".to_string(),
+
+        (_, _, minutes, _) if minutes > 1 => format!("{} minutes ago", minutes),
+        (_, _, minutes, _) if minutes == 1 => "one minute ago".to_string(),
```
Comment (jyn514) [inline src/web/mod.rs:417]:
```suggestion
        (_, _, 1, _) => "one minute ago".to_string(),
```

File: src/web/mod.rs [hunk 2]
```diff
@@ -402,25 +398,26 @@ impl Server {
...
-        format!("{} seconds ago", delta.num_seconds())
-    } else {
-        "just now".to_string()
+    match delta {
+        (days, _, _, _) if days > 5 => format!("{}", tm.strftime("%b %d, %Y").unwrap()),
+        (days, _, _, _) if days > 1 => format!("{} days ago", days),
+        (days, _, _, _) if days == 1 => "one day ago".to_string(),
+
+        (_, hours, _, _) if hours > 1 => format!("{} hours ago", hours),
+        (_, hours, _, _) if hours == 1 => "an hour ago".to_string(),
```
Comment (jyn514) [inline src/web/mod.rs:414]:
```suggestion
        (_, 1, _, _) => "an hour ago".to_string(),
```

File: src/web/mod.rs [hunk 3]
```diff
@@ -402,25 +398,26 @@ impl Server {
...
-    } else if delta.num_minutes() == 1 {
-        "one minute ago".to_string()
-    } else if delta.num_seconds() > 0 {
-        format!("{} seconds ago", delta.num_seconds())
-    } else {
-        "just now".to_string()
+    match delta {
+        (days, _, _, _) if days > 5 => format!("{}", tm.strftime("%b %d, %Y").unwrap()),
+        (days, _, _, _) if days > 1 => format!("{} days ago", days),
+        (days, _, _, _) if days == 1 => "one day ago".to_string(),
```
Comment (jyn514) [inline src/web/mod.rs:411]:
```suggestion
        (1, _, _, _) => "one day ago".to_string(),
```

Comment (jyn514) [inline src/web/mod.rs:411]:
```suggestion
        (1, _, _, _) => "one day ago".to_string(),
```

File: src/web/mod.rs [hunk 4]
```diff
@@ -402,25 +398,26 @@ impl Server {
...
-        format!("{} minutes ago", delta.num_minutes())
-    } else if delta.num_minutes() == 1 {
-        "one minute ago".to_string()
-    } else if delta.num_seconds() > 0 {
-        format!("{} seconds ago", delta.num_seconds())
-    } else {
-        "just now".to_string()
+    match delta {
+        (days, _, _, _) if days > 5 => format!("{}", tm.strftime("%b %d, %Y").unwrap()),
+        (days, _, _, _) if days > 1 => format!("{} days ago", days),
```
Comment (jyn514) [inline src/web/mod.rs:410]:
```suggestion
        (2..=5, _, _, _) => format!("{} days ago", days),
```

File: src/web/mod.rs [hunk 5]
```diff
@@ -307,13 +302,14 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
...
-    for version in &versions_sem {
-        if req_sem_ver.matches(&version.0) {
-            return Some(MatchVersion {
-                corrected_name,
-                version: MatchSemver::Semver((version.0.to_string(), version.1)),
-            });
-        }
+    if let Some((version, id)) = versions_sem
+        .iter()
+        .find(|(vers, _)| req_sem_ver.matches(vers))
```
Comment (jyn514) [inline src/web/mod.rs:307]: What's the reason for this change? I think it was more readable before.

File: src/web/mod.rs [hunk 6]
```diff
@@ -278,27 +277,23 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
...
 
     // we need to sort versions first
     let versions_sem = {
-        let mut versions_sem: Vec<(Version, i32)> = Vec::new();
+        let mut versions_sem: Vec<(Version, i32)> = Vec::with_capacity(versions.len());
 
         for version in &versions {
-            // in theory a crate must always have a semver compatible version
-            // but check result just in case
+            // in theory a crate must always have a semver compatible version, but check result just in case
```
Comment (jyn514) [inline src/web/mod.rs:296]: What's the reason for this change? It's now over 100 columns.

File: src/web/mod.rs [hunk 7]
```diff
@@ -278,27 +277,23 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
...
+        });
     }
 
     // Now try to match with semver
     let req_sem_ver = VersionReq::parse(&req_version).ok()?;
 
     // we need to sort versions first
     let versions_sem = {
-        let mut versions_sem: Vec<(Version, i32)> = Vec::new();
+        let mut versions_sem: Vec<(Version, i32)> = Vec::with_capacity(versions.len());
```
Comment (jyn514) [inline src/web/mod.rs:293]: Let's leave this for #692 .

File: src/web/mod.rs [hunk 8]
```diff
@@ -278,27 +277,23 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
...
-    // we can't expect users to use semver in query
-    for version in &versions {
-        if version.0 == req_version {
-            return Some(MatchVersion {
-                corrected_name,
-                version: MatchSemver::Exact(version.clone()),
-            });
-        }
+    // first check for exact match, we can't expect users to use semver in query
+    if let Some((version, id)) = versions.iter().find(|(vers, _)| vers == &req_version) {
```
Comment (jyn514) [inline src/web/mod.rs:281]: Same as below, I don't see the reason for using `find` or for moving the comment onto one line.

File: src/web/mod.rs [hunk 9]
```diff
@@ -243,14 +243,11 @@ impl MatchSemver {
...
-    // need to decode first
+    // version is an Option<&str> from router::Router::get, need to decode first
     use url::percent_encoding::percent_decode;
+
     let req_version = version
-        .and_then(|v| match percent_decode(v.as_bytes()).decode_utf8() {
-            Ok(p) => Some(p),
-            Err(_) => None,
-        })
+        .and_then(|v| percent_decode(v.as_bytes()).decode_utf8().ok())
```
Comment (jyn514) [inline src/web/mod.rs:250]: :heart:

File: src/web/mod.rs [hunk 10]
```diff
@@ -243,14 +243,11 @@ impl MatchSemver {
 /// underscores (`_`) and vice-versa. The return value will indicate whether the crate name has
 /// been matched exactly, or if there has been a "correction" in the name that matched instead.
 fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option<MatchVersion> {
-    // version is an Option<&str> from router::Router::get
-    // need to decode first
```
Comment (jyn514) [inline src/web/mod.rs:247]: Why move this comment onto a single line?

Comment (Kixiron) [reply src/web/mod.rs:247]: There's not really a reason for it to be two, it's one continuous thought

File: src/web/mod.rs (hunk 6 above)
Comment (Kixiron) [reply src/web/mod.rs:296]: There's not really a reason for it to be two, it's one continuous thought

File: src/web/mod.rs (hunk 5 above)
Comment (Kixiron) [reply src/web/mod.rs:307]: We have potentially better optimization from iterators and it's more idiomatic

File: src/web/mod.rs (hunk 8 above)
Comment (Kixiron) [reply src/web/mod.rs:281]: We have potentially better optimization from iterators and it's more idiomatic

File: src/web/mod.rs (hunk 4 above)
Comment (jyn514) [reply src/web/mod.rs]: Whoops, this again doesn't compile. Try

```suggestion
        (days @ 2..=5, _, _, _) => format!("{} days ago", days),
```
